learn/react-native/integrating-crashlytics.md

```diff
@@ -0,0 +1,158 @@
+
+Crashlytics can be employed for the following reasons:
+
+- Crash Reporting
+- Error Insights
+- Real time Alerts
+- Custom Logs
+
+## Using Firebase Crashlytics
+To send crash logs to Crashlytics, you need to integrate the Crashlytics SDK into your Wavemaker application's codebase.
+If you don't have a Firebase project, create one on the Firebase Console: [console.firebase.google.com](https://console.firebase.google.com/)
+Once you have created a Firebase project, you can follow the steps below to integrate Crashlytics into a WaveMaker application.
+
+### Adding Crashlytics Plugin to an App
+
+Crashlytics plugins can be installed in a few steps in a WaveMaker application. Please refer to this [page](https://docs.wavemaker.com/learn/react-native/third-party-expo-plugins#expo)
+on how to install the plugin.
+
+
+1. Install and setup the app module.
+
+```
+npm i @react-native-firebase/app
+```
+
+2. Install the Crashlytics module
+
+```
+npm i @react-native-firebase/crashlytics
+```
+
+3. Once installed, you need to add the following config plugin to your `app.json` or `app.config.json`.
+    - `@react-native-firebase/app`
+    - `@react-native-firebase/crashlytics` 
+
+**`app.json`**
+
+```javascript
+{
+  "expo": {
+    "plugins": [
+    "@react-native-firebase/app",
+    "@react-native-firebase/crashlytics"
+    ]
+    ... your existing configuration 
+  }
+}
+```
+
+### Create Firebase JSON in Expo
+
+Create `firebase.json` file in the root directory of your Wavemaker application with the Crashlytics-related keys set to the following values: 
+
+**`firebase.json`**
+
+```javascript
+{
+    "react-native":{
+    "crashlytics_auto_collection_enabled": true,
+    "crashlytics_debug_enabled": true,
+    "crashlytics_ndk_enabled": true,
+    "crashlytics_is_error_generation_on_js_crash_enabled": true,
+    "crashlytics_javascript_exception_handler_chaining_enabled": true
+    }
+}
+```
+
+### Add PreBuild Script
+
+Add `prebuild` script to `package.json`.
+
+```javascript
+{
+  "scripts": {
+       ... your existing scripts,
+        "Prebuild":"expo prebuild –clean"
+  }
+}
+```
+
+### Crashlytics Interfaces
+
+Crashlytics provides the following interfaces to log messages, events, and errors.
+
+| Methods | Description |
+| ------- | ------- |
+| log(message: String) | Log a custom message or event |
+| logException(exception: Throwable) | Logs an exception or error that occurred in the app. |
+| recordError(error: Throwable) | Records an error without causing the app to crash. |
+| setAttribute(key: String, value: Any) | Sets a custom key-value attribute that provides additional context for crash reports. |
+| setUserId(userId: String) | Associates a user identifier with crash reports, helping to identify affected users. |
+| catchException(exception: Throwable) | Catches an exception to provide more context about app issues. |
+
+
+### Crash Logs Attributes and Error Reports
```

Comment:
Lets address how to setup WM app to detect crashes using Crashlytics first and then as addon in the next section talk about reporting more errors and logs. Reporting errors and logs is not required if i just wanted crash detection, right?